### ISSUE TITLE
Allow deploying a subdirectory to Azure.

### ISSUE THREAD

**Issue by Zoltu** [travis-ci/travis-ci#6075]
I have a JavaScript project where I can run `gulp build` to create a `dist` folder containing everything needed to serve my website.  I want to be able to deploy this folder to an Azure WebApp, but it doesn't appear the Azure deploy lets me specify a single directory to deploy and instead it attempts to deploy the entire git repo (which can't be served in its bare form).

I would like an additional setting for the Azure deployment that allows me to specify a folder to deploy.

**Comment by BanzaiMan** [comment]
Is this possible at all with Azure's git deploy? How do you do this from your local machine?

**Comment by Zoltu** [comment]
There is the Azure web deploy stuff, or FTP deploy.  If you are constrained to git deploy though, you can do a git deploy of a particular folder by creating a new local repo with just the folder in question and then pushing that to the Azure git repo for that advice app service.

**Comment by aaronwolski** [comment]
Does the documentation at https://github.com/projectkudu/kudu/wiki/Customizing-deployments#deploying-a-specific-folder-for-a-node-php-or-aspnet-site apply to you?

**Comment by Zoltu** [comment]
I don't think so.  I believe when doing an Azure Deploy with TravisCI it isn't doing a `git` deploy so a custom deployment file wouldn't help.  And as I mentioned in my previous comment, doing a `git` deploy doesn't work in this scenario since the files needed to be deployed are generated as an artifact of the TravisCI build process.

**Comment by aaronwolski** [comment]
From the documentation at [Azure Web App Deployment](https://docs.travis-ci.com/user/deployment/azure-web-apps), TravisCI does use `git` to deploy to Azure. So the approach is for Travis to push a commit to the Azure local repo, which triggers a deployment to the app service.

To make this work you will need to use the `skip_cleanup` deployment option, as described in the **Pre-deploy Steps** section at [Using Travis to Deploy Apps to Azure](http://www.felixrieseberg.com/using-travis-to-deploy-apps-to-azure/), to include the build artefacts when pushing to Azure. You'll also need to include the `.deployment` file to identify the folder to deploy to the app service in the root of your repository, as detailed in my previous comment.

This should achieve what you want without a single change to TravisCI.

**Comment by Zoltu** [comment]
Ah, I didn't realize that it used git to deploy.  It sounds like what you described will work.  Unfortunately, I don't use Travis CI anymore (or Azure Web Sites for that matter) so it'll be up to someone else to verify.  Would be nice if the docs were updated so future users can benefit from this knowledge.

**Comment by vivmaha** [comment]
@aaronwolski Where can I find information on the `.deployment` file that you've mentioned? The link you have is now broken.

**Comment by paultbarham** [comment]
Has anyone here tried to deploy a Java ROOT.war file using Travis CI to an Azure WebApp?  I've got everything configured as I think I should, and Travis is not reporting any error yet my app is not being deployed (as evidenced by the Azure portal log).

Here is my yml file:

language: java
sudo: false 
script: mvn clean verify
before_deploy: 
- git config user.email "pabarham@microsoft.com"
- git config user.name "Paul T. Barham"
- git commit -am "dummy"
deploy: 
provider: azure_web_apps
verbose: true 
skip_cleanup: true 
on: 
all_branches: true 

I have environment variables defined in Travis for username, password and WebApp name.  The maven build is successful, thus generating target/ROOT.war, which I then commit to the Travis git repo with the git commands.

Any clue?

Thanks,
Paul
